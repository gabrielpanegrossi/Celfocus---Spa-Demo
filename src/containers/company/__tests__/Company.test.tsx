import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import Company from '../index';
import { QueryClientProvider, QueryClient } from 'react-query';
import * as MockReactRouterDom from 'react-router-dom';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual<typeof MockReactRouterDom>('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

jest.mock('~services', () => ({
  companies: {
    fetchOne: () => ({ id: 1, name: 'Local Public Office', vatin: '500754012' }),
  },
  numbers: {
    fetchAll: () => [{ id: '351910000000', type: 'mobile', company_id: 1 }],
  },
}));

describe('ContainerHeader', () => {
  beforeEach(() => {
    const queryClient = new QueryClient();

    render(
      <Router>
        <QueryClientProvider client={queryClient}>
          <Company />
        </QueryClientProvider>
      </Router>
    );
  });

  it('should match snpashot', () => {
    const queryClient = new QueryClient();

    const container = render(
      <Router>
        <QueryClientProvider client={queryClient}>
          <Company />
        </QueryClientProvider>
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render', () => {
    expect(screen.queryByRole('table')).toBeInTheDocument();
  });

  it('should render fetched info on table', async () => {
    expect(screen.queryByRole('table')).toBeInTheDocument();

    await waitFor(() => {
      expect(screen.queryByText('Local Public Office')).toBeInTheDocument();
      expect(screen.queryByText('351 910000000')).toBeInTheDocument();
    });
  });

  it('should navigate to company page', async () => {
    const user = userEvent.setup();

    await waitFor(() => {
      expect(screen.queryByText('351 910000000')).toBeInTheDocument();
    });

    user.click(screen.queryByText('351 910000000'));

    await waitFor(() => {
      expect(global.window.location.href).toContain('/number/351910000000');
    });
  });

  it('should navigate back', async () => {
    const user = userEvent.setup();

    user.click(screen.queryByRole('button', { name: 'Back' }));

    await waitFor(() => {
      expect(mockNavigate).toBeCalledWith(-1);
    });
  });
});
