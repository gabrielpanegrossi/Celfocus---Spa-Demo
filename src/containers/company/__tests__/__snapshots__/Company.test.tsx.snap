// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerHeader should match snpashot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-jSMfEi kDHLLL"
      >
        <header
          class="sc-bczRLJ cbMPzK"
        >
          <div
            class="sc-gsnTZi jaAxkF"
          >
            <button
              class="sc-dkzDqf"
            >
              Back
            </button>
          </div>
          <div
            class="sc-gsnTZi jaAxkF"
          >
            <h1
              class="sc-hKMtZM byoMjW"
            />
          </div>
          <div
            class="sc-gsnTZi jaAxkF"
          />
        </header>
        <div
          class="sc-gKXOVf bQoaVo"
        >
          <table
            class="sc-eCYdqJ jhWBFm"
          >
            <thead>
              <tr>
                <th>
                  <span
                    aria-busy="true"
                    aria-live="polite"
                  >
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                  </span>
                </th>
                <th>
                  <span
                    aria-busy="true"
                    aria-live="polite"
                  >
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                  </span>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  <span
                    aria-busy="true"
                    aria-live="polite"
                  >
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                  </span>
                </td>
                <td>
                  <span
                    aria-busy="true"
                    aria-live="polite"
                  >
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <div>
      <div
        class="sc-jSMfEi kDHLLL"
      >
        <header
          class="sc-bczRLJ cbMPzK"
        >
          <div
            class="sc-gsnTZi jaAxkF"
          >
            <button
              class="sc-dkzDqf"
            >
              Back
            </button>
          </div>
          <div
            class="sc-gsnTZi jaAxkF"
          >
            <h1
              class="sc-hKMtZM byoMjW"
            />
          </div>
          <div
            class="sc-gsnTZi jaAxkF"
          />
        </header>
        <div
          class="sc-gKXOVf bQoaVo"
        >
          <table
            class="sc-eCYdqJ jhWBFm"
          >
            <thead>
              <tr>
                <th>
                  <span
                    aria-busy="true"
                    aria-live="polite"
                  >
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                  </span>
                </th>
                <th>
                  <span
                    aria-busy="true"
                    aria-live="polite"
                  >
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                  </span>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  <span
                    aria-busy="true"
                    aria-live="polite"
                  >
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                  </span>
                </td>
                <td>
                  <span
                    aria-busy="true"
                    aria-live="polite"
                  >
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                    <span
                      class="react-loading-skeleton"
                      style="width: 100%;"
                    >
                      ‌
                    </span>
                    <br />
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-jSMfEi kDHLLL"
    >
      <header
        class="sc-bczRLJ cbMPzK"
      >
        <div
          class="sc-gsnTZi jaAxkF"
        >
          <button
            class="sc-dkzDqf"
          >
            Back
          </button>
        </div>
        <div
          class="sc-gsnTZi jaAxkF"
        >
          <h1
            class="sc-hKMtZM byoMjW"
          />
        </div>
        <div
          class="sc-gsnTZi jaAxkF"
        />
      </header>
      <div
        class="sc-gKXOVf bQoaVo"
      >
        <table
          class="sc-eCYdqJ jhWBFm"
        >
          <thead>
            <tr>
              <th>
                <span
                  aria-busy="true"
                  aria-live="polite"
                >
                  <span
                    class="react-loading-skeleton"
                    style="width: 100%;"
                  >
                    ‌
                  </span>
                  <br />
                </span>
              </th>
              <th>
                <span
                  aria-busy="true"
                  aria-live="polite"
                >
                  <span
                    class="react-loading-skeleton"
                    style="width: 100%;"
                  >
                    ‌
                  </span>
                  <br />
                </span>
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                <span
                  aria-busy="true"
                  aria-live="polite"
                >
                  <span
                    class="react-loading-skeleton"
                    style="width: 100%;"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                    style="width: 100%;"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                    style="width: 100%;"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                    style="width: 100%;"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                    style="width: 100%;"
                  >
                    ‌
                  </span>
                  <br />
                </span>
              </td>
              <td>
                <span
                  aria-busy="true"
                  aria-live="polite"
                >
                  <span
                    class="react-loading-skeleton"
                    style="width: 100%;"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                    style="width: 100%;"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                    style="width: 100%;"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                    style="width: 100%;"
                  >
                    ‌
                  </span>
                  <br />
                  <span
                    class="react-loading-skeleton"
                    style="width: 100%;"
                  >
                    ‌
                  </span>
                  <br />
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
